btr-web/btr-main-app/utils/btr-bods/bods-to-si-converters.ts

```diff
@@ -99,29 +100,65 @@ const _getSiPerson = (btrBodsPerson: BtrBodsPersonI): ProfileI => {
   }
 }
 
-const _getMaxInterestValue = (oocs: BtrBodsOwnershipOrControlI, interestType: BodsInterestTypeE): number => {
-  let max = 0
-
-  let maxCalc = (oldValue: number, newValue: number): number => oldValue + newValue
-
-  if (oocs.source?.description === BtrSourceDescriptionProvidedByBtrGovBC) {
-    maxCalc = (currentValue: number, newValue: number): number => {
-      return Math.max(currentValue, newValue)
-    }
-  }
+// const _getMaxInterestValue = (oocs: BtrBodsOwnershipOrControlI, interestType: BodsInterestTypeE): number => {
+//   let max = 0
+
+//   let maxCalc = (oldValue: number, newValue: number): number => oldValue + newValue
+
+//   if (oocs.source?.description === BtrSourceDescriptionProvidedByBtrGovBC) {
+//     maxCalc = (currentValue: number, newValue: number): number => {
+//       return Math.max(currentValue, newValue)
+//     }
+//   }
+
+//   for (const interest of oocs.interests) {
+//     if (interest.type === interestType) {
+//       max = maxCalc(max,
+//         interest.share?.exact ||
+//         interest.share?.maximum ||
+//         interest.share?.minimum ||
+//         0
+//       )
+//     }
+//   }
+
+//   return max > 100 ? 100 : max
+// }
+
+const _getPercentageRange = (oocs: BtrBodsOwnershipOrControlI, interestType: BodsInterestTypeE): PercentageRangeE => {
+  /**
+   * Note: we assume the min and max values for all interests in oocs.interests are the same.
+   * The first occurance of the interest that matches the interestType will be used to determine the range.
+   *
+   * We need to confirm:
+   * 1) for oocs (ownershipOrControlStatement) provided by BtrSourceDescriptionProvidedByBtrGovBC,
+   *    is it possible for two interests to have different min and max values?
+   *
+   * 2) how do we want to handle the oocs from external sources?
+   */
+  let min: number | undefined = 0
+  let max: number | undefined = 0
+  let range: PercentageRangeE = PercentageRangeE.NO_SELECTION
 
   for (const interest of oocs.interests) {
     if (interest.type === interestType) {
-      max = maxCalc(max,
-        interest.share?.exact ||
-        interest.share?.maximum ||
-        interest.share?.minimum ||
-        0
-      )
+      min = interest.share?.minimum
+      max = interest.share?.maximum
+      break
     }
   }
 
-  return max > 100 ? 100 : max
+  if (min === 0 && max === 25) {
+    range = PercentageRangeE.LESS_THAN_25
+  } else if (min === 25 && max === 50) {
+    range = PercentageRangeE.BETWEEN_25_AND_50
+  } else if (min === 50 && max === 75) {
+    range = PercentageRangeE.BETWEEN_50_AND_75
+  } else if (min === 75 && max === 100) {
+    range = PercentageRangeE.MORE_THAN_75
+  }
```

Comment:
please add an 'else' here that logs out an error including the business identifier. That way we will know its an issue with that business's data and our monitoring will pick it up